### PR TITLE
Use conda-forge-pinnings for readline

### PR DESCRIPTION
**Pull request by saraedum**

### CODE REVIEW

**Comment by saraedum** [comment]
@conda-forge-admin please rerender

**Comment by saraedum** [comment]
@conda-forge-admin please rerender

**Comment by saraedum** [comment]
I don't understand AppVeyor's complaint:
```
The build phase is set to "MSBuild" mode (default), but no Visual Studio project or solution files were found in the root directory. If you are not building Visual Studio project switch build mode to "Script" and provide your custom build command.
```

Also, I am not sure what to do about `ecl`. Is it a build/run dependency?

**Comment by saraedum** [comment]
Hm…I have `skip: true  # [win]`. Why does appveyor even bother to build this?

**Comment by saraedum** [comment]
@conda-forge-admin please rerender

**Comment by saraedum** [comment]
@conda-forge-admin please rerender

**Comment by saraedum** [comment]
@isuruf @embray: Not sure why appveyor tried to build here but other than that this looks good to me.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,18 +30,18 @@ source:
     - patches/maxima.system.patch
 
 build:
-  number: 2
+  number: 3
   skip: true  # [win]
 
 requirements:
   build:
-    - toolchain
```
Comment (isuruf) [inline recipe/meta.yaml]: I think you need a compiler as `ecl` is a source-to-source compiler which translates lisp to c.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -30,18 +30,18 @@ source:
...
     - libtool
     - texinfo
-    - readline 6.2*
+  host:
     - ecl >=6.1.3-1
+    - readline
```
Comment (isuruf) [inline recipe/meta.yaml]: Can you add,
```yaml
  host:
    - gmp
    - bdw-gc
  run:
    - gmp
    - {{ pin_compatible('bdw-gc') }}
```

Comment (saraedum) [reply recipe/meta.yaml]: Ok.

**Pull request merged**